### PR TITLE
Fix: mounting state use theme

### PR DESCRIPTION
**Pull request by harshsinghatz**
## What does this PR do?
- This PR fixes the rendering issue that was causing the referenced issue below

Getting the unmount state seems redundant after cleanup seems redundant as the main use of this is to show a text only till the isReady is true

Fixes # (issue)
Reference: https://github.com/calcom/cal.com/issues/7191

**Environment**: Staging(main branch) / Production

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

**Comment by harshsinghatz** [comment]
@hariombalhara

File: packages/lib/hooks/useMountedState.ts [hunk 1]
```diff
@@ -9,13 +9,12 @@ import { useCallback, useEffect, useRef } from "react";
 export default function useMountedState(): () => boolean {
   const mountedRef = useRef<boolean>(false);
```
Comment (hariombalhara) [inline packages/lib/hooks/useMountedState.ts:10]: Do we really need this ref now?

File: packages/lib/hooks/useMountedState.ts [hunk 2]
```diff
@@ -9,13 +9,12 @@ import { useCallback, useEffect, useRef } from "react";
...
   const get = useCallback(() => mountedRef.current, []);
+  const [mount, setMount] = useState(false);
```
Comment (hariombalhara) [inline packages/lib/hooks/useMountedState.ts:12]: Better name woud be mounted and setMounted

File: packages/lib/hooks/useMountedState.ts (hunk 1 above)
Comment (harshsinghatz) [reply packages/lib/hooks/useMountedState.ts]: No, removed it!

File: packages/lib/hooks/useMountedState.ts (hunk 2 above)
Comment (harshsinghatz) [reply packages/lib/hooks/useMountedState.ts]: makes sense, changes made

**Review by G3root — CHANGES_REQUESTED** [review]
`IsMounted` value returned by useTheme hook is only used in the above select component, adding a hook with a setState call can cause unnecessary re-renders in other components which doesn't need the `isMounted` value. It's safe to use the `isMounted` only inside the select component and remove it from the useTheme hook.

**Comment by harshsinghatz** [comment]
> `IsMounted` value returned by useTheme hook is only used in the above select component, adding a hook with a setState call can cause unnecessary re-renders in other components. It's safe to use the `isMounted` only inside the select component and remove it from the useTheme hook.

I don't think the re-renders would be an issue, the only time it would re-render the component is when component is actually mounted to replace the placeholder component. 
but if we don't want it i have removed the hook

**Comment by harshsinghatz** [comment]
@G3root ?

**Comment by G3root** [comment]
> > `IsMounted` value returned by useTheme hook is only used in the above select component, adding a hook with a setState call can cause unnecessary re-renders in other components. It's safe to use the `isMounted` only inside the select component and remove it from the useTheme hook.
> 
> I don't think the re-renders would be an issue, the only time it would re-render the component is when component is actually mounted to replace the placeholder component. but if we don't want it i have removed the hook @G3root

updated my previous comment to be more clear.

**Review by G3root — APPROVED** [review]
LGTM :)  Awesome work ❤️

**Review by hariombalhara — APPROVED** [review]

**Pull request merged**